torch/distributed/optim/post_localSGD_optimizer.py

```diff
@@ -65,10 +66,19 @@ def __repr__(self):
         return self.optim.__repr__()
 
     def state_dict(self):
-        return self.optim.state_dict()
+        optim_state_dict = self.optim.state_dict()
```

Comment:
Add a docstring mentioning it is the same as optim.Optimizer state_dict, but adds model averager's `step` to the checkpoint to ensure reload does not cause unnecessary warm up again?

Reply:
good point, added this to an updated version